### PR TITLE
Simplify engine contract lookup query

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
Follow up to #4538, simplifies the query run to lookup contracts for interpretation.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/JdbcLedgerDao.scala [hunk 1]
```diff
@@ -1728,32 +1728,18 @@ object JdbcLedgerDao {
...
-         |  cd.id,
-         |  cd.contract,
-         |  c.transaction_id,
-         |  c.create_event_id,
-         |  c.workflow_id,
-         |  c.key,
-         |  le.effective_at,
-         |  string_agg(distinct sigs.signatory, '$PARTY_SEPARATOR') as signatories,
-         |  string_agg(distinct obs.observer, '$PARTY_SEPARATOR') as observers
+         |select cd.contract
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/JdbcLedgerDao.scala]: I guess without the string aggregation the query could even be moved out of the database specific traits?

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/JdbcLedgerDao.scala]: For sure, I just wanted to solve one issue at a time and keep the PR as simple as possible.

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/JdbcLedgerDao.scala]: Tracked by https://github.com/digital-asset/daml/issues/4544

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/JdbcLedgerDao.scala]: Addressed in https://github.com/digital-asset/daml/pull/4545

**Pull request merged**